### PR TITLE
HLA-1342: Documentation for S_REGION code change

### PR DESCRIPTION
**Pull request by s-goldman**
Resolves [HLA-1342](https://jira.stsci.edu/browse/HLA-1342)

This PR adds additional documentation describing the recent changes to the determination of the S_REGIONS. I've included animations/gifs created by @rlwastro and hosted on https://ssb.stsci.edu/alignment/. 

**Checklist for maintainers**
- [x] added entry in `CHANGELOG.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Comment by s-goldman** [comment]
For ease of reading, here is a pdf of the changes in the html format. 
[Single-visit Mosaic Processing — DrizzlePac 3.6.2rc1.dev86+g4c07c9d9.d20240711 documentation.pdf](https://github.com/user-attachments/files/17210292/Single-visit.Mosaic.Processing.DrizzlePac.3.6.2rc1.dev86%2Bg4c07c9d9.d20240711.documentation.pdf)

File: doc/source/mast_data_products/svm_processing.rst [hunk 1]
```diff
@@ -653,32 +653,77 @@ Defining the Footprint
...
+that build up a total mask (extract_mask), and find the vertices/corners of that mask
+(find_corners).
+
+Dilation and Erosion
+********************
+We use the mathematical morphology operations of binary erosion and binary dilation
+in different orders for two purposes. Using erosion and then dilation removes pixels
+and then inflates the outer region. This has the effect of isolating island pixels and
+removing noise. Doing the opposite, erosion and then dilation, will result in more-unique
+peninsulas and edge features being saved but smoothed to some degree.
```
Comment (rlwastro) [inline doc/source/mast_data_products/svm_processing.rst:680]:
```suggestion
in different orders for two purposes. Using erosion and then dilation (the morphological
"opening" operation) removes isolated pixels and narrow tendrils.  
This has the effect of isolating islands of pixels that are thinly connected to one another and
removing noise. Doing the opposite, dilation and then erosion (the morphological "closing"
operation), fills in small holes and gaps to join islands and smooth the outer edge
to some degree.
```

File: doc/source/mast_data_products/svm_processing.rst [hunk 2]
```diff
@@ -653,32 +653,77 @@ Defining the Footprint
...
+
+Extracting a Mask
+*****************
+Extracting the the mask first involves using a ndimage.binary_fill_holes to fill
+in any holes surrounded by data. The code then uses dilation to keep edge features
+while smoothing the outer edge. Erosion is then used to get a mask that has a similar
+shape and size to the original mask. We then combine the resulting mask with the original
+mask to ensure we retain edge features while yielding a smoother edge.
```
Comment (rlwastro) [inline doc/source/mast_data_products/svm_processing.rst:688]:
```suggestion
mask to ensure we retain features that touch the image edges.
```

File: doc/source/mast_data_products/svm_processing.rst [hunk 3]
```diff
@@ -653,32 +653,77 @@ Defining the Footprint
...
+
+
+Finding Border Pixels
+*********************
+The code traces the edge of the footprint to isolate only the edge pixels. It
+then uses erosion and then dilation to remove noise and island pixels. It then
+applies a gaussian filter to the edge pixels to ultimately create a smoother S_REGION.
+The resulting edge pixels are then simplified.
```
Comment (rlwastro) [inline doc/source/mast_data_products/svm_processing.rst:696]:
```suggestion
To find all the edge pixels, an eroded version of the mask is subtracted from the mask,
which leaves behind only the pixels around the edge.  A box search algorithm is used to 
connect the edge pixels in a path going counter-clockwise around the image edge.
Finally, the resulting edge path is simplified.
```

File: doc/source/mast_data_products/svm_processing.rst [hunk 4]
```diff
@@ -653,32 +653,77 @@ Defining the Footprint
...
+
+
+Simplifying the Border
+**********************
+In order to decrease the number of vertices for our S_REGION we use the simplify-polyline
+package. A call to simplify-polyline.simplify looks at each set of three
+consecutive vertices and determines if the errors in the line changes significantly
+when the middle pixel is removed. If it is below an set error (min_dist), the code
+removes the pixel. It should be noted that these three pixels can be far apart.
```
Comment (rlwastro) [inline doc/source/mast_data_products/svm_processing.rst:705]:
```suggestion
removes the pixel. This algorithm is repeated until no more pixels are found to remove.
It should be noted that in later iterations, these three pixels can be far apart.
```

**Review by rlwastro — APPROVED** [review]
Made some clarifying comments, mostly things look good.

**Pull request merged**